crates/ruff_linter/src/rules/pycodestyle/rules/multiple_imports_on_one_line.rs

```diff
@@ -27,17 +34,88 @@ use crate::checkers::ast::Checker;
 pub struct MultipleImportsOnOneLine;
 
 impl Violation for MultipleImportsOnOneLine {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
     #[derive_message_formats]
     fn message(&self) -> String {
         format!("Multiple imports on one line")
     }
+
+    fn fix_title(&self) -> Option<String> {
+        Some(format!("Split imports"))
+    }
 }
 
 /// E401
 pub(crate) fn multiple_imports_on_one_line(checker: &mut Checker, stmt: &Stmt, names: &[Alias]) {
     if names.len() > 1 {
-        checker
-            .diagnostics
-            .push(Diagnostic::new(MultipleImportsOnOneLine, stmt.range()));
+        let mut diagnostic = Diagnostic::new(MultipleImportsOnOneLine, stmt.range());
+        if checker.settings.preview.is_enabled() {
+            diagnostic.set_fix(split_imports(
+                stmt,
+                names,
+                checker.locator(),
+                checker.indexer(),
+                checker.stylist(),
+            ));
+        }
+        checker.diagnostics.push(diagnostic);
+    }
+}
+
+/// Generate a [`Fix`] to split the imports across multiple statements.
+fn split_imports(
+    stmt: &Stmt,
+    names: &[Alias],
+    locator: &Locator,
+    indexer: &Indexer,
+    stylist: &Stylist,
+) -> Fix {
+    if indexer.in_multi_statement_line(stmt, locator) {
+        // Ex) `x = 1; import os, sys` (convert to `x = 1; import os; import sys`)
+        let replacement = names
+            .iter()
+            .map(|alias| {
+                let Alias {
+                    range: _,
+                    name,
+                    asname,
+                } = alias;
+
+                if let Some(asname) = asname {
+                    format!("import {name} as {asname}")
+                } else {
+                    format!("import {name}")
+                }
+            })
+            .join("; ");
+
+        Fix::safe_edit(Edit::range_replacement(replacement, stmt.range()))
```

Comment:
@diceroll123 - I decided to just `; `-join these for now, since it's hard to know if we're _allowed_ to split them.